cpp/include/raft/core/logger.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/logger_macros.hpp>
+
+#include <rapids_logger/logger.hpp>
+
+#include <sstream>
+
+namespace raft {
+
+/**
+ * @brief Returns the default sink for the global logger.
+ *
+ * If the environment variable `RAFT_DEBUG_LOG_FILE` is defined, the default sink is a sink to that
+ * file. Otherwise, the default is to dump to stderr.
+ *
+ * @return sink_ptr The sink to use
+ */
+inline rapids_logger::sink_ptr default_sink()
+{
+  auto* filename = std::getenv("RAFT_DEBUG_LOG_FILE");
+  return (filename == nullptr)
+           ? static_cast<rapids_logger::sink_ptr>(std::make_shared<rapids_logger::stderr_sink_mt>())
```

Comment:
Same question as on cudf about `static_cast`.

Reply:
This is fixed now.

Reply:
Fixed.

Reply:
Fixed in https://github.com/rapidsai/raft/pull/2566/commits/c817c0c586ad55f237e72bd278b6cfac5afb61ac